src/dictionary.rs

```diff
@@ -237,10 +237,13 @@ pub trait AsyncBuiltins<'forth, T: 'static> {
     /// for each builtin name. See [the `AsyncBuiltin` trait's
     /// documentation][impling] for details on implementing this method.
     ///
+    /// The `id` parameter is `'static`, as we will only resolve function names
+    /// from the provided [`Self::BUILTINS`], which must be `'static`.
+    ///
     /// [`Future`]: core::future::Future
     /// [`match`]: https://doc.rust-lang.org/stable/std/keyword.match.html
     /// [impling]: #implementing-async-builtins
-    fn dispatch_async(&self, id: &FaStr, forth: &'forth mut crate::Forth<T>) -> Self::Future;
+    fn dispatch_async(&self, id: &'static FaStr, forth: &'forth mut crate::Forth<T>) -> Self::Future;
```

Comment:
TIOLI: we could also just give this the less-restrictive `'forth` lifetime, which would also be sufficient to allow it to be moved into the returned `Future` --- the future only needs to live for `'forth`, as it is not spawned, but run inside the VM's `process_line`. Since the function names are only resolved from the builtins array, we _can_ make this `'static`, but just making it `'forth` would potentially allow us more flexibility in the future.

Up to you!

Reply:
Yeah, I considered that as well, though I felt `'static` is more *correct* as at least right now, all `FaStr`s that get provided here originate from the `&'static [AsyncBuiltin]` array, NOT the bump allocator that would be indicated with `'forth`.

If we need to break it/change it to `'forth` later, I'm okay with breaking it then!